src/common/com/intellij/plugins/haxe/lang/psi/HaxeClassResolveResult.java

```diff
@@ -365,6 +365,12 @@ public void specializeByParameters(@Nullable List<PsiElement> typeList) {
       final PsiElement specializedType = typeList.get(i);
 
       if (genericParamName == null) continue;
+
+      // HACK to avoid resolving Dynamic<Dynamic> when expected value is just Dynamic
+      //quote from docs: "Dynamic is a special type because it allows explicit declaration with and without a type parameter."
+      // current implementation would see that Dynamic<T> has a type parameter
+      if(specializedType!= null && "Dynamic".equals(specializedType.getText())) continue;
+
```

Comment:
What is the code path (stack) that required this hack?  I know that this was fixed in a number of situations by using a different resolve call that passed a Specialization.  I just wonder if this would be fixed by resolving through that path, instead of needing the hack.

Reply:
without this one line all tests except `testNoErrorOnConstrainedGenericOverrides` will run fine, however this one test will show the following  implementation as mismatching
```haxe
public function copy()<error descr="Incompatible return type: IMap<Class<Dynamic>, V> should be IMap<Class<Dynamic<Dynamic>>, V>">:IMap<K, V></error> {}
```
now I could perhaps  just handle this in the `canAssign` part of the code, `Dynamic` and `Dynamic<Dynamic>` should be the same after all, but any errors involving this type would look strange as it would say `Dynamic<Dynamic>` instead of just `Dynamic`.

The quick fix was  this hack,  as  any use of type Parameter `Dynamic` could  end up being resolved to `Dynamic<T>` and then to `dynamic<Dynamic>`

Reply:
Hmmm.. The canAssign part of the code _should_ be working here.  The call at line 374 calls the proper code path that I've validated in the past (and I contend should be working correctly here).  I may have to debug this to understand why it's not working as I expect.  I don't care a whole lot about showing "Dynamic<Dynamic>" on screen; it's technically correct, even if a bit weird looking.